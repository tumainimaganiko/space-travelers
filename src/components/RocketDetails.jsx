import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import styles from '../styles/RocketDetails.module.css';
import { bookRocket, cancelRocket } from '../redux/rockets/rocketsSlice';

const RocketDetails = ({
  name, description, image, id, reserved,
}) => {
  const dispatch = useDispatch();

  return (
    <div className={styles.container}>
      <div>
        <img src={image} className={styles.img} alt="" />
      </div>
      <div className={styles.details}>
        <h2>{name}</h2>
        <p>
          {reserved && <span>reserved</span>}
          {description}
        </p>
        {reserved ? (
          <button
            type="button"
            className={styles.cancelledButton}
            onClick={() => dispatch(cancelRocket(id))}
          >
            Cancel Reservation
          </button>
        ) : (
          <button
            type="button"
            className={styles.reservedButton}
            onClick={() => dispatch(bookRocket(id))}
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </div>
  );
};

export default RocketDetails;

RocketDetails.propTypes = PropTypes.shape({
  name: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.arrayOf(PropTypes.string),
  id: PropTypes.string,
  reserved: PropTypes.bool,
}).isRequired;
