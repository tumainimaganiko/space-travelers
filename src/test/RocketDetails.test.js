import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import RocketDetails from '../components/RocketDetails';
import myRockets from './rocketData';

const reducer = (
  state = {
    rockets: { rocket: myRockets },
  },
) => state;

const store = configureStore({ reducer });

it('Should render Rocket details', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <RocketDetails
          name="space 1"
          description="first space"
          image={['image1', 'image2']}
          id="firstItem"
          reserved
        />
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
