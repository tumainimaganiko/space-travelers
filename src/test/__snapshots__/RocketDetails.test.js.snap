// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Rocket details 1`] = `
<div
  className="container"
>
  <div>
    <img
      alt=""
      className="img"
      src={
        Array [
          "image1",
          "image2",
        ]
      }
    />
  </div>
  <div
    className="details"
  >
    <h2>
      space 1
    </h2>
    <p>
      <span>
        reserved
      </span>
      first space
    </p>
    <button
      className="cancelledButton"
      onClick={[Function]}
      type="button"
    >
      Cancel Reservation
    </button>
  </div>
</div>
`;
